dokka-subprojects/analysis-kotlin-symbols/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/symbols/translators/DefaultSymbolToDocumentableTranslator.kt

```diff
@@ -886,7 +887,24 @@ internal class DokkaSymbolVisitor(
 
 
     // ----------- Translators of modifiers ----------------------------------------------------------------------------
-    private fun KtSymbolWithModality.getDokkaModality() = modality.toDokkaModifier()
+    private fun KtSymbolWithModality.getDokkaModality(): KotlinModifier {
+        val isInterface = this is KtClassOrObjectSymbol && classKind == KtClassKind.INTERFACE
+        return if (isInterface) {
+            when (modality) {
+                // modifiers other than "sealed" are redundant for interfaces
+                Modality.SEALED -> KotlinModifier.Sealed
+                else -> KotlinModifier.Empty
+            }
+        } else {
+            when (modality) {
+                Modality.FINAL -> KotlinModifier.Final
+                Modality.SEALED -> KotlinModifier.Sealed
+                Modality.OPEN -> KotlinModifier.Open
+                Modality.ABSTRACT -> KotlinModifier.Abstract
+                else -> KotlinModifier.Empty
```

Comment:
`else` branch is redundant here.
An exhaustive `when` can help to track the changes.